Remove 'recurse' parameter from Inline.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26703 Remove convert_to_ssa argument from runCleanupPasses - it is only used in one place.
* #26705 Use optimized_graph in graph_executor.
* #26489 Use optimized graph in Inline (essentially, making Inline recursive now).
* #26488 Add 'optimized_graph' to Function.
* #26571 Use std::mutex instead of std::call_once in Function when we initialize GraphExecutor.
* **#26487 Remove 'recurse' parameter from Inline.**

The way it is implemented currently is bad because while we're inlining
to a graph G, we are also mutating all the graphs that are being
inlined. The problem is that the graphs we're inlining are usually the
original graphs of functions, so we're silently changing them behind the
scenes, and we don't have a way to recover 'unoptimized' graphs
afterwards.

Differential Revision: [D17485748](https://our.internmc.facebook.com/intern/diff/D17485748)

Commit:
Remove 'recurse' parameter from Inline.

The way it is implemented currently is bad because while we're inlining
to a graph G, we are also mutating all the graphs that are being
inlined. The problem is that the graphs we're inlining are usually the
original graphs of functions, so we're silently changing them behind the
scenes, and we don't have a way to recover 'unoptimized' graphs
afterwards.

[ghstack-poisoned]

houseroad commented on torch/onnx/utils.py:
```diff
@@ -87,7 +87,7 @@ def _split_tensor_list_constants(g, block):
 
 def _optimize_graph(graph, operator_export_type, _disable_torch_constant_prop=False):
     # Inline everyting (recursively)
```
Do you want to update the comment as well?

ZolotukhinM replied:
Oh, right, thanks (it will become recursive again in #26489, but I think we can just omit that word from the comment).

Commit:
Update on "Remove 'recurse' parameter from Inline."


The way it is implemented currently is bad because while we're inlining
to a graph G, we are also mutating all the graphs that are being
inlined. The problem is that the graphs we're inlining are usually the
original graphs of functions, so we're silently changing them behind the
scenes, and we don't have a way to recover 'unoptimized' graphs
afterwards.

Differential Revision: [D17485748](https://our.internmc.facebook.com/intern/diff/D17485748)

[ghstack-poisoned]

Commit:
Update on "Remove 'recurse' parameter from Inline."


The way it is implemented currently is bad because while we're inlining
to a graph G, we are also mutating all the graphs that are being
inlined. The problem is that the graphs we're inlining are usually the
original graphs of functions, so we're silently changing them behind the
scenes, and we don't have a way to recover 'unoptimized' graphs
afterwards.

Differential Revision: [D17485748](https://our.internmc.facebook.com/intern/diff/D17485748)

[ghstack-poisoned]

jerryzh168 commented on test/cpp/jit/test_inliner.cpp:
```diff
@@ -39,26 +39,14 @@ struct InlinerGuard {
+    Inline(*g);
     FileCheck()
         .check("three")
         ->check("two")
```
so `foo2` is inlined and `foo1` is not inlined right? then should you `check_not` one as well?

jerryzh168 replied:
oh, I saw check_count("prim::CallFunction", 1), but I think still better to add a "check_not" for "one" since that is what you are checking for previously

ZolotukhinM replied:
We check the number of  `CallFunction` entries, if we inline everything it would be 0. What you suggest is another possible way to verify the same, but I don't think there is a reason to modify the existing test.

ZolotukhinM replied:
This test has been checking non-recursive inlining before and I just kept it as-is :) I just removed the test for recursive inlining.

jerryzh168 replied:
Yeah, I saw this test is changed to check recursive inlining in later PRs.

Commit:
Update on "Remove 'recurse' parameter from Inline."


The way it is implemented currently is bad because while we're inlining
to a graph G, we are also mutating all the graphs that are being
inlined. The problem is that the graphs we're inlining are usually the
original graphs of functions, so we're silently changing them behind the
scenes, and we don't have a way to recover 'unoptimized' graphs
afterwards.

Differential Revision: [D17485748](https://our.internmc.facebook.com/intern/diff/D17485748)

[ghstack-poisoned]
